### PR TITLE
gh-115538: Remove PYTHONHOME when running test_venv.test_multiprocessing_recursion

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-115538

### CODE REVIEW

File: Lib/test/test_venv.py [hunk 1]
```diff
@@ -531,7 +531,8 @@ def test_multiprocessing_recursion(self):
         rmtree(self.env_dir)
         self.run_with_capture(venv.create, self.env_dir)
         script = os.path.join(TEST_HOME_DIR, '_test_venv_multiprocessing.py')
-        subprocess.check_call([self.envpy(real_env_dir=True), script])
+        subprocess.check_call([self.envpy(real_env_dir=True), script],
+                              env={**os.environ, "PYTHONHOME": ""})
```
Comment (vstinner) [inline Lib/test/test_venv.py]: Only setting PYTHONHOME sounds a little bit too specific. I suggest to use the isolated option instead, ``-I``:

```suggestion
        subprocess.check_call([self.envpy(real_env_dir=True), "-I", script])
```

Comment (aisk) [reply Lib/test/test_venv.py]: Thanks for the review! I test it locally and it can resolve the issue too.

**Review by vstinner — APPROVED** [review]
LGTM. Using -I isolated mode in Python test suite is always a good thing.

**Comment by aisk** [comment]
> @aisk: Can you please you please try to backport this change to 3.11 and 3.12 branches?

Certainly, done.

**Pull request merged**